core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -627,6 +641,29 @@ private[spark] class ExecutorAllocationManager(
     def isExecutorIdle(executorId: String): Boolean = {
       !executorIdToTaskIds.contains(executorId)
     }
+
+    /**
+     * Get the number of locality aware pending tasks and related locality preferences as the
+     * hints used for executor allocation.
+     */
+    def executorPlacementHints(): (Int, Map[String, Int]) =
+      allocationManager.synchronized {
+      var localityAwarePendingTasks: Int = 0
+      val localityToCount = new mutable.HashMap[String, Int]()
+      stageIdToPreferredLocations.values.foreach { localities =>
```

Comment:
Hi @sryza , that's a good question. Reason why I didn't differentiate pending, running and completed tasks is that what we get here is a list to locality preferences `Seq[Seq[String]]`, where we don't have task id mapping information (we cannot get the exact task id at that point in DAGScheduler), so it is hard to differentiate the status with such fine-grained way.